features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5xGattServer.cpp

```diff
@@ -451,6 +514,140 @@ void nRF5xGattServer::hwCallback(ble_evt_t *p_ble_evt)
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    // start the copy of what is in input
+                    memcpy(req->data + req->length, input_req.data, input_req.len);
+
+                    // update the lenght of the data written
+                    req->length = req->length + input_req.len;
+
+                    // success, signal it to the softdevice
+                    ble_gatts_rw_authorize_reply_params_t reply = {
+                        .type = BLE_GATTS_AUTHORIZE_TYPE_WRITE,
+                        .params = {
+                            .write = {
+                                .gatt_status = BLE_GATT_STATUS_SUCCESS,
+                                .update = 1,
+                                .offset = input_req.offset,
+                                .len = input_req.len,
+                                .p_data = input_req.data
+                            }
+                        }
+                    };
+
+                    sd_ble_gatts_rw_authorize_reply(conn_handle, &reply);
+                }   return;
+
+                case BLE_GATTS_OP_EXEC_WRITE_REQ_CANCEL: {
+                    releaseLongWriteRequest(conn_handle);
+                    sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_succes_reply);
+                }   return;
+
+                case BLE_GATTS_OP_EXEC_WRITE_REQ_NOW: {
+                    long_write_request_t* req = findLongWriteRequest(conn_handle);
+                    if (!req) {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_reply);
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    GattWriteAuthCallbackParams cbParams = {
+                        .connHandle = conn_handle,
+                        .handle     = req->attr_handle,
+                        .offset     = req->offset,
+                        .len        = req->length,
+                        .data       = req->data,
+                        .authorizationReply = AUTH_CALLBACK_REPLY_SUCCESS /* the callback handler must leave this member
+                                                                           * set to AUTH_CALLBACK_REPLY_SUCCESS if the client
+                                                                           * request is to proceed. */
+                    };
+                    uint16_t write_authorization = p_characteristics[characteristicIndex]->authorizeWrite(&cbParams);
+
+                    // the user code didn't provide the write authorization,
+                    // just leave here.
+                    if (write_authorization != AUTH_CALLBACK_REPLY_SUCCESS) {
+                        // report the status of the operation in any cases
+                        sd_ble_gatts_rw_authorize_reply(gattsEventP->conn_handle, &write_auth_invalid_reply);
```

Comment:
It would be better to use the local "conn_handle", as in other calls (for better maintainability).